dispatch/modules/content/models.py

```diff
@@ -401,14 +431,50 @@ def save(self, **kwargs):
         super(Image, self).save(**kwargs)
 
         if is_new and self.img:
-            image = Img.open(StringIO.StringIO(self.img.read()))
+
+            d = self.img.read()
+
+            xmp_start = d.find('<x:xmpmeta')
+            xmp_end = d.find('</x:xmpmeta')
+            #if the image contains XMP data, parse the relevant info
+            if xmp_start != -1:
+                xmp_str = d[xmp_start:xmp_end]
+                print xmp_str
+                self.title = self.find_xmp("title", xmp_str)
+                self.caption = self.find_xmp("description", xmp_str)
```

Comment:
These lines will set `title` and `caption` to empty strings (`''`) if they aren't set in the XML. It's probably a good idea to add a check to prevent this